Formula/node@18.rb

```diff
@@ -0,0 +1,108 @@
+class NodeAT18 < Formula
+  desc "Platform built on V8 to build network applications"
+  homepage "https://nodejs.org/"
+  url "https://nodejs.org/dist/v18.12.0/node-v18.12.0.tar.xz"
+  sha256 "73a7f01e2999eb197763ced666a6cd544ad580eaefb73e0a849603b3e804f42e"
+  license "MIT"
+
+  livecheck do
+    url "https://nodejs.org/dist/"
+    regex(%r{href=["']?v?(18(?:\.\d+)+)/?["' >]}i)
+  end
+
+  keg_only :versioned_formula
+
+  # https://nodejs.org/en/about/releases/
+  # disable! date: "2025-04-30", because: :unsupported
+  deprecate! date: "2023-10-18", because: :unsupported
+
+  depends_on "pkg-config" => :build
+  depends_on "python@3.10" => :build
+  depends_on "brotli"
+  depends_on "c-ares"
+  depends_on "icu4c"
+  depends_on "libnghttp2"
+  depends_on "libuv"
+  depends_on "openssl@1.1"
+
+  uses_from_macos "python", since: :catalina
+  uses_from_macos "zlib"
+
+  on_macos do
+    depends_on "llvm" => [:build, :test] if DevelopmentTools.clang_build_version <= 1100
+  end
+
+  fails_with :clang do
+    build 1100
+    cause <<~EOS
+      error: calling a private constructor of class 'v8::internal::(anonymous namespace)::RegExpParserImpl<uint8_t>'
+    EOS
+  end
+
+  fails_with gcc: "5"
+
+  def install
+    ENV.llvm_clang if OS.mac? && (DevelopmentTools.clang_build_version <= 1100)
+
+    # make sure subprocesses spawned by make are using our Python 3
+    ENV["PYTHON"] = which("python3.10")
+
+    args = %W[
+      --prefix=#{prefix}
+      --with-intl=system-icu
+      --shared-libuv
+      --shared-nghttp2
+      --shared-openssl
+      --shared-zlib
+      --shared-brotli
+      --shared-cares
+      --shared-libuv-includes=#{Formula["libuv"].include}
+      --shared-libuv-libpath=#{Formula["libuv"].lib}
+      --shared-nghttp2-includes=#{Formula["libnghttp2"].include}
+      --shared-nghttp2-libpath=#{Formula["libnghttp2"].lib}
+      --shared-openssl-includes=#{Formula["openssl@1.1"].include}
+      --shared-openssl-libpath=#{Formula["openssl@1.1"].lib}
+      --shared-brotli-includes=#{Formula["brotli"].include}
+      --shared-brotli-libpath=#{Formula["brotli"].lib}
+      --shared-cares-includes=#{Formula["c-ares"].include}
+      --shared-cares-libpath=#{Formula["c-ares"].lib}
+      --openssl-use-def-ca-store
+    ]
+
+    system "./configure", *args
+    system "make", "install"
+  end
+
+  def post_install
+    (node_modules/"npm/npmrc").atomic_write("prefix = #{HOMEBREW_PREFIX}\n")
```

Comment:
The variable `node_modules` is not defined.

```suggestion
    (lib/"node_modules/npm/npmrc").atomic_write("prefix = #{HOMEBREW_PREFIX}\n")
```